snuba/subscriptions/store.py

```diff
@@ -0,0 +1,104 @@
+import json
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Collection
+
+from rediscluster import StrictRedisCluster
```

Comment:
Why not using the client from `snuba.redis` (redis_client)?
that decides how to setup the cluster for you (if you need a cluster) according to config.

Reply:
Discussed this with @tkaemming beforehand and we'd prefer to pass `redis_client` into the `RedisSubscriptionStore`, rather than import a global. This type is probably wrong though now that I think about it, although I'm not sure what would be right since I guess there's no common base class.